Content.Server/Storage/EntitySystems/EntityStorageSystem.cs

```diff
@@ -138,17 +138,18 @@ public void OpenStorage(EntityUid uid, EntityStorageComponent? component = null)
 
     public void CloseStorage(EntityUid uid, EntityStorageComponent? component = null)
     {
-        if (!Resolve(uid, ref component))
+        var xform = Transform(uid);
+        if (!Resolve(uid, ref component) || xform.GridUid is not { } grid)
             return;
         component.Open = false;
 
-        var targetCoordinates = new EntityCoordinates(uid, component.EnteringOffset);
-
-        var entities = _lookup.GetEntitiesInRange(targetCoordinates, component.EnteringRange, LookupFlags.Approximate);
+        var translation = xform.WorldMatrix.Transform(component.EnteringOffset) - xform.WorldPosition;
+        var reduced = _lookup.GetWorldAABB(uid, xform).Translated(translation).Enlarged(component.EnteringRange);
+        var entities = _lookup.GetEntitiesIntersecting(grid, reduced, LookupFlags.Approximate);
```

Comment:
This is still real bad, just slap a TODO on it for me to expose collisionmanager so we can get proper collisions and I'll deal with it in 3-6 months.